src/v/thirdparty/protobuf/BUILD

```diff
@@ -0,0 +1,16 @@
+#
+# TODO(bazel) this is a temporary hack to fix layering check violation with
+# protobuf v24 until we upgrade to v26, which doesn't seem to have the problem,
+# but can't immediately move to v26 without dealing with some depcreated
+# interfaces. The `.h` file is not listed anywhere as a public header, and the
+# `.pb.h` beats me. all of the other proto files work as expected.
+#
+cc_library(
+    name = "descriptor",
+    hdrs = [
+        "descriptor.h",
+        "descriptor.pb.h",
```

Comment:
Fixes for layering check violations. This one is still a bit mysterious to me. Well, for `descriptor.h` it's straight forward: in v24 this doesn't seem to be exported through a publicly visible library. For the `pb.h` one it's weird. Even if I create an explicit proto_cc_library for it, it doesn't seem to see it.

In any case, none of this is necessary when we move to protobuf v26, but i ran into other compat issues there as we need to re-implement some bits that rely on fully deprecated/removed interfaces.

Reply:
I'm going to have to upgrade to the latest protobuf for the protobuf parser so I can dig into this.

Reply:
>I'm going to have to upgrade to the latest protobuf for the protobuf parser so I can dig into this.

Amazing. I briefly looked at the the replacement interface for the deprecated bits and it didn't look too cray.

Reply:
https://github.com/BenPope/redpanda/tree/update_libprotobuf

I didn't have time to look at disabling the logging.

Reply:
https://github.com/protocolbuffers/protobuf/issues/16914

Reply:
> [protocolbuffers/protobuf#16914](https://github.com/protocolbuffers/protobuf/issues/16914)

Yeah, that's where I stopped.